docs/en/Running-Hive-with-Alluxio.md

```diff
@@ -72,6 +72,112 @@ FIELDS TERMINATED BY '|'
 LOCATION 'alluxio://master_hostname:port/ml-100k';
 ```
 
+## Create Internal Table from Files Located in Alluxio
+
+Hive can create internal tables from files stored on Alluxio and Hive will manage the lifecycle of internal tables.
+When you drop an internal table, Hive drops the table metadata and data both.
+
+### Hive CLI examples
+
+Use the data file in `ml-100k.zip` from
+[http://grouplens.org/datasets/movielens/](http://grouplens.org/datasets/movielens/) as an example.
+
+```bash
+$ bin/alluxio fs mkdir /ml-100k
+$ bin/alluxio fs copyFromLocal /path/to/ml-100k/u.user alluxio://master_hostname:port/ml-100k
+```
+
+Then create an internal table:
+
+```
+hive> CREATE TABLE u_user (
+userid INT,
+age INT,
+gender CHAR(1),
+occupation STRING,
+zipcode STRING)
+ROW FORMAT DELIMITED
+FIELDS TERMINATED BY '|'
+LOCATION 'alluxio://master_hostname:port/ml-100k';
+```
+
+## Use Alluxio as the FileSystem for External tables currently in HDFS
+
+### Hive CLI examples
+
+Create an external table in HDFS:
+
+```
+hive> CREATE EXTERNAL TABLE u_user (
+userid INT,
+age INT,
+gender CHAR(1),
+occupation STRING,
+zipcode STRING)
+ROW FORMAT DELIMITED
+FIELDS TERMINATED BY '|'
+LOCATION 'hdfs://namenode_hostname:port/ml-100k';
+```
+
+We assume that you have set `alluxio.underfs.address=hdfs://namenode:port/` in `alluxio-site.properties`.
+
+Use the following HiveQL to change the table data location：
+
+```
+hive> alter table TABLE_NAME set location "alluxio://master_hostname:port/ml-100k";
+```
+
+Use the following HiveQL and check the "Location" attribute to verify whether the table location is set correctly:
+
+```
+hive> desc formatted u_user;
+```
+
+## Use Alluxio as the FileSystem for Internal tables currently in HDFS
+
+### Hive CLI examples
+
+We assume that the **"hive.metastore.warehouse.dir"** property is set as following: 
+```xml
+<property>
+  <name>hive.metastore.warehouse.dir</name>
+  <value>/user/hive/warehouse</value>
+</property>
+```
+
+Create an internal table in HDFS and load data into it:
+
+```
+hive> CREATE TABLE u_user (
+userid INT,
+age INT,
+gender CHAR(1),
+occupation STRING,
+zipcode STRING)
+ROW FORMAT DELIMITED
+FIELDS TERMINATED BY '|';
+
+hive> LOAD DATA LOCAL INPATH '/path/to/ml-100k/u.user'
+OVERWRITE INTO TABLE u_user;
+```
+
+We assume that you have set `alluxio.underfs.address=hdfs://namenode:port/` in `alluxio-site.properties`.
+
+Use the following HiveQL to change the table data location：
+
+```
+hive> alter table TABLE_NAME set location "alluxio://master_hostname:port/user/hive/warehouse";
```

Comment:
This should be `alluxio://master_hostname:port/user/hive/warehouse/u_user` where `u_user` should be the internal table name in hive, right ? otherwise it does not work on my side